Marlin/src/module/probe.cpp

```diff
@@ -505,7 +530,7 @@ float Probe::run_z_probe(const bool sanity_check/*=true*/) {
     #else
       UNUSED(plbl);
     #endif
-    return probe_fail || early_fail;
+    return (int)(probe_fail || early_fail);
```

Comment:
You could try making the lambda's return type explicitly a bool,. I'm not sure why this isn't deducing it correctly. Every return statement I see was a bool.

I believe the syntax for the return type would be:

`auto try_to_probe = [&](...argument list...) -> bool { ... body ... }`

Reply:
tried just casting to a bool for now to see if its happy and have it matched to the rest.